walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,74 @@
 
 package com.google.sps;
 
+import java.io.*;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+
 
 public final class FindMeetingQuery {
+  // array representing the number of meetings happening during that minute
+  private ArrayList<Integer> meetings;
+
+  public FindMeetingQuery() {
+    meetings = new ArrayList<Integer>();
+    //before processing the events, the number of meetings for each minute is 0
+    meetings.addAll(Collections.nCopies(TimeRange.END_OF_DAY + 2, 0));
+  }
+
+  /** returns a Collection of time ranges when meeting {@code request} can be scheduled in the day of 
+    * events so that all attendees are free */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      // if the meeting lasts more than a day, there is no solution
+      return new ArrayList<TimeRange>();
+    }
+
+    Collection<String> attendees = request.getAttendees();
+
+    for (Event event : events) {
+      // at least one requested attendee is participating in the event
+      if (event.containsRequestedAttendees(attendees)) {
+        // update the number of meetings during the time of event
+        updateNumberOfMeetings(event.getWhen());
+      }
+    }
+
+    return findAvailableTimeRanges((int)request.getDuration());
+  }
```

Comment:
I have no idea if this will work ... I guess the next step is the tests to make sure that it does right? :smiling_imp:

Reply:
Looks like there are already tests written for this function.

Ana, can you confirm you ran the tests and they all pass?

Reply:
Yes, all the tests passed.